nautobot/dcim/api/serializers.py

```diff
@@ -1025,8 +1025,14 @@ def validate(self, data):
                 )
 
         # Validate many-to-many VLAN assignments
+        tagged_vlans = data.get("tagged_vlans", [])
+        if tagged_vlans and data.get("mode") != InterfaceModeChoices.MODE_TAGGED:
```

Comment:
This doesn't work on a `PATCH` to add `tagged_vlans` when `mode` is not specified in the request:

```json
{
  "id": "03d0892a-4b1e-4956-bade-792a3ead7643",
  "display": "Ethernet0",
  "url": "http://localhost:8080/api/dcim/interfaces/03d0892a-4b1e-4956-bade-792a3ead7643/",
  "device": {
    "display": "rtr1",
    "id": "9a3cf918-b8da-40eb-9883-cb51b63ef40e",
    "url": "http://localhost:8080/api/dcim/devices/9a3cf918-b8da-40eb-9883-cb51b63ef40e/",
    "name": "rtr1"
  },
  "name": "Ethernet0",
  "label": "",
  "type": {
    "value": "1000base-t",
    "label": "1000BASE-T (1GE)"
  },
  "enabled": true,
  "parent_interface": null,
  "bridge": null,
  "lag": null,
  "mtu": null,
  "mac_address": null,
  "mgmt_only": false,
  "description": "",
  "mode": {
    "value": "tagged",
    "label": "Tagged"
  },
  "untagged_vlan": null,
  "tagged_vlans": [],
  "cable": null,
  "cable_peer": null,
  "cable_peer_type": null,
  "connected_endpoint": null,
  "connected_endpoint_type": null,
  "connected_endpoint_reachable": null,
  "count_ipaddresses": 0,
  "created": "2022-10-27",
  "last_updated": "2022-10-27T13:29:45.009498Z",
  "tags": [],
  "notes_url": "http://localhost:8080/api/dcim/interfaces/03d0892a-4b1e-4956-bade-792a3ead7643/notes/",
  "custom_fields": {}
}
```


```
curl -X 'PATCH' \
  'http://localhost:8080/api/dcim/interfaces/03d0892a-4b1e-4956-bade-792a3ead7643/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: MaDmbYcv6DjjmlxvfSQockE4q2Svb5pACqya1DWcoOFw7oDtLM49jx8FYdlhEBVS' \
  -d '{
  "tagged_vlans": [
    "ae4a4270-9e68-4040-8000-9da70767f039"
  ]
}'

Response:
{
  "tagged_vlans": [
    "Mode must be set to tagged when specifying tagged_vlan"
  ]
}
```